node/staking.go

```diff
@@ -20,56 +22,38 @@ import (
 //Refer: https://solidity.readthedocs.io/en/develop/abi-spec.html
 
 const (
-	// DepositFuncSignature is the func signature for deposit method
-	DepositFuncSignature  = "0xd0e30db0"
-	withdrawFuncSignature = "0x2e1a7d4d"
-	funcSingatureBytes    = 4
+	funcSingatureBytes = 4
+	lockPeriodInEpochs = 3 // This should be in sync with contracts/StakeLockContract.sol
 )
 
-// UpdateStakingList updates the stakes of every node.
-// TODO: read directly from smart contract, or at least check the receipt also for incompleted transaction.
-func (node *Node) UpdateStakingList(block *types.Block) error {
-	signerType := types.HomesteadSigner{}
-	txns := block.Transactions()
-	for i := range txns {
-		txn := txns[i]
-		toAddress := txn.To()
-		if toAddress != nil && *toAddress != node.StakingContractAddress { //Not a address aimed at the staking contract.
-			utils.GetLogInstance().Info("Mismatched Staking Contract Address", "expected", node.StakingContractAddress.Hex(), "got", toAddress.Hex())
-			continue
-		}
-		currentSender, _ := types.Sender(signerType, txn)
-		_, isPresent := node.CurrentStakes[currentSender]
-		data := txn.Data()
-		switch funcSignature := decodeFuncSign(data); funcSignature {
-		case DepositFuncSignature: //deposit, currently: 0xd0e30db0
-			amount := txn.Value()
-			value := amount.Int64()
-			if isPresent {
-				//This means the node has increased its stake.
-				node.CurrentStakes[currentSender] += value
-			} else {
-				//This means its a new node that is staking the first time.
-				node.CurrentStakes[currentSender] = value
+// StakeInfo is the struct for the return value of listLockedAddresses func in stake contract.
+type StakeInfo struct {
+	LockedAddresses  []common.Address
+	BlockNums        []*big.Int
+	LockPeriodCounts []*big.Int
+	Amounts          []*big.Int
+}
+
```

Comment:
It will be helpful to add comments to what `LockPeriodCounts` means. Could we change `Amounts` to `Stakes`?

Reply:
I know its there in the contract itself ;) But a comment will help.

Reply:
Done.